app/main.py

```diff
@@ -1,3 +1,64 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | float | int) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
```

Comment:
If you don't want to return a TypeError, it means you expect the inputs to be either Distance, float, or int. Therefore, the elif is redundant in this case. You can simply return Distance(self.km + other). Please make this adjustment throughout your code.

Reply:
remove all lines  the elife ?

Reply:
if isinstance(other, Distance):
    return Distance(self.km + other.km)
return Distance(self.km + other)

Reply:
I already understood, but I did as you originally asked